internal/blocksync/reactor.go

```diff
@@ -184,21 +187,21 @@ func (r *Reactor) OnStop() {
 // respondToPeer loads a block and sends it to the requesting peer, if we have it.
 // Otherwise, we'll respond saying we do not have it.
 func (r *Reactor) respondToPeer(ctx context.Context, msg *bcproto.BlockRequest, peerID types.NodeID, blockSyncCh *p2p.Channel) error {
-	block := r.store.LoadBlock(msg.Height)
+	block := r.store.LoadBlockProto(msg.Height)
```

Comment:
In a previous version, where I was reading the commit in addition to the block (what is now done when reading the extending commit), when I had both being returned as Objects and then converted to Proto again, the tests were constantly timing out. In the meantime I removed the commit but somehow it made sense to me to only read Proto instead of converting it back and forth so I left this change. This logic should probably be applied for the extended commit now that it is also read. Especially if the extended commit ends up being larger, we might have a problem again.

Reply:
> the tests were constantly timing out

That seems strange. And you were able to track it to changing the struct.

Reply:
Yes, I left it like this. I think in any case it is more elegant to not convert it back and forth when not needed.